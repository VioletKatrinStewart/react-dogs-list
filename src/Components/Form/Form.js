import React from 'react';

export default function Form({ name, bio, age, image, breed, handleChange, handleSave }) {
  return (
    <div className="form">
      <h2> Enter the details to add a dog </h2>
      <label>
        Name
        <input
          value={name}
          onChange={(e) => {
            handleChange('name', e.target.value);
          }}
          type="text"
          placeholder="Name"
        />
      </label>
      <label>
        Bio
        <input
          value={bio}
          onChange={(e) => {
            handleChange('bio', e.target.value);
          }}
          type="text"
          placeholder="Bio"
        />
      </label>
      <label>
        Age
        <input
          value={age}
          onChange={(e) => {
            handleChange('age', e.target.value);
          }}
          type="number"
          placeholder="Age"
        />
      </label>
      <label>
        Image
        <input
          value={image}
          onChange={(e) => {
            handleChange('image', e.target.value);
          }}
          type="text"
          placeholder="image"
        />
      </label>
      <label>
        Breed
        <input
          value={breed}
          onChange={(e) => {
            handleChange('breed', e.target.value);
          }}
          type="text"
          placeholder="Breed"
        />
      </label>

      <button onClick={handleSave}>Submit</button>
    </div>
  );
}
