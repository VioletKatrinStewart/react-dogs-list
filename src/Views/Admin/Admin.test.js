import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import Form from './Admin';

it('should render text in form', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/admin']}>
      {' '}
      <Route path="/admin">
        <Form />
      </Route>
    </MemoryRouter>
  );
  await screen.findByText('Name');
  expect(container).toMatchSnapshot();
});
