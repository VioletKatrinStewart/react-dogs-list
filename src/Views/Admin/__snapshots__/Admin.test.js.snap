// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render text in form 1`] = `
<div>
   
  <div
    class="admindiv"
  >
    <header>
      <a
        href="/"
      >
        Home
      </a>
      <a
        aria-current="page"
        class="active"
        href="/admin"
      >
        Admin
      </a>
    </header>
    <div
      class="form"
    >
      <h2>
         Enter the details to add a dog 
      </h2>
      <p />
      <label>
        Name
        <input
          placeholder="Name"
          type="text"
          value=""
        />
      </label>
      <label>
        Bio
        <input
          placeholder="Bio"
          type="text"
          value=""
        />
      </label>
      <label>
        Age
        <input
          placeholder="Age"
          type="number"
          value=""
        />
      </label>
      <label>
        Image
        <input
          placeholder="image"
          type="text"
          value=""
        />
      </label>
      <label>
        Breed
        <input
          placeholder="Breed"
          type="text"
          value=""
        />
      </label>
      <button>
        Submit
      </button>
    </div>
  </div>
</div>
`;
